### DISCUSSION TITLE
DeePMD potential for thermal conductivity calculation

### DISCUSSION THREAD

**Discussion by ZhihaoXu0313** [deepmodeling/deepmd-kit#1914] [Q&A]
Hi,

I'm trying to calculate the thermal conductivity with the trained potential. If I don't want to use EMD simulation to do that, for example, I'd like to use ShengBTE to calculate TCs with 2nd order force constant and 3rd order force constant, I can use phonoLammps combining with my trained potential to get 2nd order force constant. 

But how can I get the 3rd order constant? It's there any software can calculate IFC3 with deepmd potential like what I did for IFC-2? I will appreciate it if anyone has any idea or experience. Thanks a lot.

Sincerely,
ZX

**Comment by wanghan-iapcm** [comment]
One may use finite difference to approximate the 3rd order energy derivatives. I really not sure about the accuracy, because, the dps are only trained to reproduce the first order derivatives (forces) of the energy.

**Reply by ZhihaoXu0313** [reply]
Yes, I agree with you. It seems that we could only use the NNPs to relax the structure and then throw it back to classical DFT calculations to obtain IFC3. Anyway, thanks for your reply.

ZX

**Comment by Jingtong-Zhang** [comment]
Actually the way to obtain 2rd force constants with DP is the same with that to obtain 3rd force constants.
After you have used thirdorder and generate hundreds (or even thousands) of 'POSCAR', you only need to change them into lammps file, and run MD simulation to get forces with your DP model. I don't know if there are any software can do so, but I have write a code to do so.
